### DISCUSSION TITLE
Service visit in Visited dashboard

### DISCUSSION THREAD

**Discussion by krezac** [teslamate-org/teslamate#1984] [Ideas]
**Describe the bug**

The service visit represented by latlon (37.394287,-122.150604) is not displayed in Drives, but it's still visible in Visited. It causes unzooming of the map.

**Expected behavior**

"Service mode" points (37.394287,-122.150604) excluded from Visited dashboard.
Looks like simple exclusion in dashboard definition is enough, but I'm not sure about possible side effects
`.. WHERE latitude <> 37.394287 AND longitude <> -122.150604 AND ...`

**How to reproduce it (as minimally and precisely as possible)**:

1. have a service visit (or simulate it by inserting data into positions table)
2. open the Visited dashboard
3. you will see jump to US and back

**Relevant entries from the logs**

N/A

**Screenshots**

![Screen Shot 2021-09-07 at 12 25 33](https://user-images.githubusercontent.com/16595657/132331101-c7c5fd5a-5b69-4409-9f83-79daef90716a.png)
![Screen Shot 2021-09-07 at 12 26 37](https://user-images.githubusercontent.com/16595657/132331123-38ee7549-0f15-456e-84b4-1d40b83bf05b.png)

**Data**

[positions.csv](https://github.com/adriankumpf/teslamate/files/7120851/positions.csv)

**Environment**

- TeslaMate version: 1.24.0
- Type of installation: Docker
- OS TeslaMate is installed on: Linux docker-server 5.4.0-74-generic #83-Ubuntu SMP Sat May 8 02:35:39 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
- User OS & Browser: Mac/Chrome

**Comment by adriankumpf** [comment]
If the Tesla API falsely reports incorrect GPS positions, that's not a TeslaMate bug, right? We could implement a detection of extreme position deviations, but that wouldn't work for those who are near the falsely reported positions.

I'll move this to feature requests.

**Comment by krezac** [comment]
@adriankumpf I don't think it's "falsely reports  incorrect GPS" case (in the meaning: "bug"), from my two service visits it seems to be intentional (and deterministic) behavior of Tesla API while car is in service mode. Maybe someone else can confirm?

**Comment by SigLinJo** [comment]
I have a very edge case situation where I have had a car in Tiblisi Georgia sending data to my Tesla account for half a year until I finally managed to hunt down the owner and convince him to spoof someone else's VIN (reported to Tesla, really annoying problem) Now I am looking to clean up my database and remove everything related to locations in Tiblisi. As these situations can occur (has anyone ended up on null island maybe?) or as the thread starter who got a position in California maybe some type of process to correct this would be nice. A solution could be some help with database syntax to find and remove all coordinates that has to do with wrong locations?

**Reply by SigLinJo** [reply]
I found a solution. For me everything bogus had a longitude that was way outside what I had in any of my "real" data so `delete from positions where longitude > 40;` worked very well for me. I have my data back in order, sort of again... Lost a lot of drives when conflict between our cars made Tesla or teslamate stop listening to my car... But Just happy to be rid of all the lines going across the continent, and also all the bogus charges I had :) Once again big thank you @adriankumpf for a great peace of software!
